File: HireHubDemo/hirehub-core/src/main/java/com/hirehub/core/entities/Referral.java
package com.hirehub.core.entities;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Entity
@Table(name = "referrals")
@Data
public class Referral {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "job_id", nullable = false)
    private Long jobId;

    @Column(name = "candidate_id", nullable = false)
    private Long candidateId;

    @Column(name = "employee_id", nullable = false)
    private Long employeeId;

    @Column(name = "referral_date", nullable = false)
    private LocalDateTime referralDate;

    @Column(name = "status", nullable = false)
    private Integer status;

    // Navigation properties
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "job_id", insertable = false, updatable = false)
    private Job job;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "candidate_id", insertable = false, updatable = false)
    private Candidate candidate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id", insertable = false, updatable = false)
    private Employee employee;
}

File: HireHubDemo/HireHubDemo/hirehub-referral/Models/Referral.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HireHub.Referral.Models
{
    [Table("referrals")]
    public class Referral
    {
        [Key]
        public long Id { get; set; }

        [Column("job_id")]
        public long JobId { get; set; }

        [Column("employee_id")]
        public long EmployeeId { get; set; }

        [Column("candidate_id")]
        public long CandidateId { get; set; }

        [Column("referral_date")]
        public DateTime ReferralDate { get; set; } = DateTime.UtcNow;
    }
}
